AlpsSDK/ProximityHandler.swift

```diff
@@ -0,0 +1,233 @@
+                    let deviceId = "NEED CHANGE to a guard"
+                        sendProximityEvent(deviceId: deviceId, proximityEvent: proximityEvent) {(_ proximityEvent) in
+                            guard let pe = proximityEvent else {
+                                return
+                            }
+                            self.addProximityEvent(deviceId: id, proximityEvent: pe, clProximity: clProximity)
+                    }
+                } else {
+                    //this id is already triggered and might need to be refresh -> refreshTriggers()
+                }
+            }
+        }
+    }
+    
+    // Re-launch a proximity event depending on refreshTimer (Default : 60 seconds)
+    func refreshTriggers() {
+        for clProximity in CLProximity.allValues {
+            var (beacons, trigger, distance) = setUpTriggerBeaconsProximityEvent(forCLProximity: clProximity)
+            for id in beacons {
+                // Check if the existed proximity event needs a refresh on a based timer
+                let proximityEvent = trigger[id]
+                // Represents  the UNIX current time in milliseconds
+                let now = Int64(Date().timeIntervalSince1970 * 1000)
+                guard let proximityEventCreatedAt = proximityEvent?.createdAt else {
+                    return
+                }
+                let gap = now - proximityEventCreatedAt
+                let truncatedGap = Int(truncatingBitPattern: gap)
+                if truncatedGap > refreshTimer {
+                    // Send the refreshing proximity event based on the timer
+                    let newProximityEvent = ProximityEvent.init(deviceId: id, distance: distance)
+                    let deviceId = "NEED CHANGE"
+                    sendProximityEvent(deviceId: deviceId, proximityEvent: newProximityEvent) {(_ proximityEvent) in
+                        guard let pe = proximityEvent else {
+                            return
+                        }
+                        self.addProximityEvent(deviceId: id, proximityEvent: pe, clProximity: clProximity)
+                    }
+                } else {
+                    // Do something when it doesn't need to be refreshed
+                }
+            }
+        }
+    }
+    
+    // Remove the previous sent Proximity Event that was higher than a fixed time (Default: 5 minutes).
+    @objc
+    private func cleanTriggers() {
+        func clean(trigger: [String: ProximityEvent]) -> [String: ProximityEvent] {
+            var t = trigger
+            for (id, proximityEvent) in t {
+                guard let createdAt = proximityEvent.createdAt else {
+                    return
+                }
+                let now = Int64(Date().timeIntervalSince1970 * 1000)
+                let gap = now - createdAt
+                // If gap is higher than 5 minutes we will clear the value in the trigger dictionary
+                if gap > 5 * 60 * 1000 {
```

Comment:
let not use implicit numbers. It should some kind of variable or Date extension. For example `Date.timeIntervalFor(minutes: 5)`